broadway/api/utils/multiqueue.py

```diff
@@ -0,0 +1,45 @@
+from queue import Queue, Empty
+
+"""
+A wrapper around Python's queue module that contains multiple queues
+and round-robins between them when pulling.
+"""
+
+
+class MultiQueue:
+    def __init__(self):
+        self.queues = {}
+        self.keys = []
+        self.round_robin_idx = 0
+
+    def add_queue(self, queue_id):
+        if queue_id in self.queues:
+            raise Exception("{} already exists in the MultiQueue.".format(queue_id))
+
+        self.queues[queue_id] = Queue()
+        self.keys.append(queue_id)
+
+    def push(self, queue_id, elem):
+        if queue_id not in self.queues:
+            raise Exception("{} does not exist in the MultiQueue.".format(queue_id))
```

Comment:
What if we add a course after the API starts by manually editing the DB?
Maybe we can make this method such that if the queue_id does not exist, we create it? (we can do so fast since we are using a dict)

Reply:
Good idea, changed to that.